puppet/spec/unit/indirector/catalog/puppetdb_spec.rb

```diff
@@ -702,5 +702,55 @@ def test_file_require(resource_title, require_title)
           "catalog_uuid", "producer"]
       end
     end
+
+    describe "#redact_sensitive_params" do
+      let(:secret) {'xyzzy XKPGMKIB253ZVJHKOKZZXPQSSE'}
+      it 'should remove any sensitive resource parameters' do
+        input = {'tags'=>Puppet::Util::TagSet.new(['settings']),
```

Comment:
OK, I asked @mullr, who told me I needed the let to fix scoping wrt `secret` and `has_secret?`, and he seemed inclined.